server/controllers/authentication.coffee

```diff
@@ -93,14 +93,19 @@ module.exports.saveUnauthenticatedUser = (req, res, next) ->
         User.all (err, users) ->
             return next new Error err if err
             # if existing user document with password -> request rejected
-            if users[0]?.password
+            if users[0]?.password and not requestData.password?
```

Comment:
I don't really understand why there is another condition here with requestData. In this way, it seems that I would be able to update any other properties of the user document, except password, in an unauthenticated way, isn't it?